GVFS/GVFS.Mount/InProcessMount.cs

```diff
@@ -549,15 +550,16 @@ private void MountAndStartWorkingDirectoryCallbacks(CacheServerInfo cache)
             this.heartbeat.Start();
 
             // Launch a background job to compute the multi-pack-index. Will do nothing if up-to-date.
-            this.fileSystemCallbacks.LaunchPostFetchJob(packIndexes: new List<string>());
+            this.maintenanceJobManager.LaunchPostFetchJob(packIndexes: new List<string>());

```

Comment:
I think the callers of this new class don't really care about "jobs". We aren't even really doing anything that resembles a job, we're just doing some of the work asynchronously.

E.g. this call here could be simplified to `this.localCache.UpdateMidx`. The caller really doesn't care if that is sync or async, just that it's telling the local cache that there are potentially new packfiles available.